vyked/registry.py

```diff
@@ -381,6 +381,23 @@ def _handle_log_change(self, packet, protocol):
             handler.setLevel(level)
         protocol.send('Logging level updated')
 
+    def _handle_ping(self, packet, protocol):
+        """ Responds to pings from registry_client only if the node_ids present in the ping payload are registered
+        :param packet: The 'ping' packet received
+        :param protocol: The protocol on which the pong should be sent
+        """
+        if 'payload' in packet:
+            is_valid_node = True
+            node_ids = list(packet['payload'].values())
+            for node_id in node_ids:
+                if self._repository.get_node(node_id) is None:
+                    is_valid_node = False
+                    break
+            if is_valid_node:
+                self._pong(packet, protocol)
+        else:
+            self._pong(packet, protocol)
```

Comment:
try this maybe:

        if 'payload' in packet:
            if all(map(self.repository.get_node, packet['payload'].values())):
                self.pong(packet, protocol)
        else:
            self._pong(packet, protocol)